nucypher-core/src/dkg.rs

```diff
@@ -78,6 +92,121 @@ impl<'a> ProtocolObjectInner<'a> for ThresholdDecryptionRequest {
 
 impl<'a> ProtocolObject<'a> for ThresholdDecryptionRequest {}
 
+/// A request for an Ursula to derive a decryption share that specifies the key to encrypt Ursula's response.
+#[derive(PartialEq, Debug, Clone, Serialize, Deserialize)]
+pub struct E2EEThresholdDecryptionRequest {
```

Comment:
Just checking, does the third "E" stand for something? E2E is "end to end", right?

Reply:
✔️ E2E is what it should be since it isn't encrypted (third E) - I modified my thinking about this functionality but never updated the class name.

Reply:
I find this name to be describing something too broad that it reaches the level of inaccuracy,  The name "E2E" describes the wider-usage instead of this object instead of it's substance, and falls out of convention with other encrypted entities like `EncryptedTreasureMap`.

```suggestion
pub struct EncryptedThresholdDecryptionRequest {
```

Reply:
After a re-read I think I'm mixed up here. Can you help me understand why `EncryptedThesholdDecryptionRequest` and `E2EEThresholddecryptionRequest` need to be different?

Reply:
The crux of the design is around the ability to re-use the same `ThresholdDecryptionRequest`:

1. To be encrypted for different Ursulas - the `request_encrypting_key`s

2. To be used by different Bobs (see #3128 where Enrico returns a decryption request that can be reused by any Bob) i.e. to be associated with different `response_encrypting_key`s


Therefore, we don't want to put the `response_encrypring_key` on the `ThresholdDecryptionRequest` object itself because this would be per-Bob.


Next, the `response_encrypting_key` needs to also be encrypted. So currently, that is done by `ThresholdDecryptionRequest.encrypt(request_encrypting_key, response_encrypting_key)` and the resulting object is a `EncryptedThresholdDecryptionRequest` which holds the request and response key both encrypted using the `request_encrypting_key`. This struct only has the encrypted bytes and the ritual id.


When this object gets decrypted using `EncryptedThresholdDecryptionRequest.decrypt(secret_key)` it needs to return the decryption request (plain form) and the `response_encrypting_key`. These two items are therefore represented by this `E2EThresholdDecryptionRequest`. It has two members, `decryption_request` and `response_encrypting_key`. It's basically and intermediary object/struct to be consistent since JS does not allow tuples.


I'm open to alternative names.